core/src/main/java/com/sequenceiq/cloudbreak/controller/v4/AutoscaleV4Controller.java

```diff
@@ -132,10 +158,37 @@ public AutoscaleStackV4Response getInternalAutoscaleClusterByName(String name, @
     @InternalOnly
     public void decommissionInternalInstancesForClusterCrn(@TenantAwareParam @ResourceCrn String clusterCrn,
             List<String> instanceIds, Boolean forced) {
+        LOGGER.info("decommissionInternalInstancesForClusterCrn. forced={}, clusterCrn={}, instanceIds=[{}]",
+                forced, clusterCrn, instanceIds);
         stackCommonService.deleteMultipleInstancesInWorkspace(NameOrCrn.ofCrn(clusterCrn), restRequestThreadLocalService.getRequestedWorkspaceId(),
                 new HashSet(instanceIds), forced);
     }
 
+    @Override
+    @InternalOnly
+    public void stopInternalInstancesForClusterCrn(@TenantAwareParam @ResourceCrn String clusterCrn, @NotEmpty List<String> instanceIds,
+            Boolean forced, ScalingStrategy scalingStrategy) {
+        LOGGER.info("stopInternalInstancesForClusterCrn. ScalingStrategy={}, forced={}, clusterCrn={}, instanceIds=[{}]",
+                scalingStrategy, forced, clusterCrn, instanceIds);
+        if (scalingStrategy == null) {
+            scalingStrategy = ScalingStrategy.STOPSTART;
+            LOGGER.debug("Scaling strategy is null, and has been set to the default: {}", scalingStrategy);
+        }
+        stackCommonService.deleteMultipleInstancesInWorkspace(NameOrCrn.ofCrn(clusterCrn), restRequestThreadLocalService.getRequestedWorkspaceId(),
```

Comment:
so stopping here means deleting instances?

Reply:
I would create a separate method for stop and not mix with delete if the API is indeed aims to only stop instances

Reply:
So a different method in StackCommonService? Will make that change.

Reply:
Fixed in the next push. Resolving conversation.